flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -2203,13 +2197,24 @@ public StreamGraph getStreamGraph() {
      */
     @Internal
     public StreamGraph getStreamGraph(boolean clearTransformations) {
+        updateCacheTransformation();
```

Comment:
We can skip the check if `cachedTransformation` is empty. But I think we should still check event if all the transformations are cached, since the cache can become invalid.